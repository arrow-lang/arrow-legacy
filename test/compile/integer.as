let i8: int8 = 10;
let i16: int16 = 2000;
let i32: int32 = 300000;
let i64: int64 = 4000000000;
let i128: int128 = 50000000000000000000;

let i8: int8 = 10;
let i16: int16 = i8;
let i32: int32 = i16;
let i64: int64 = i32;
let i128: int128 = i64;

let u8: uint8 = 10;
let u16: uint16 = 2000;
let u32: uint32 = 300000;
let u64: uint64 = 4000000000;
let u128: uint128 = 50000000000000000000;

let u8: uint8 = 10;
let u16: uint16 = u8;
let u32: uint32 = u16;
let u64: uint64 = u32;
let u128: uint128 = u64;

let i = 0; // `int`
let i: int = i8;
let i: int = i16;
let i: int = i32;
let i: int = i64;
let i: int = i128;
let i: int = u8;
let i: int = u16;
let i: int = u32;
let i: int = u64;
let i: int = u128;
