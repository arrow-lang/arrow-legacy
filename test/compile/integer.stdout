@i8 = internal constant i8 10
@i16 = internal constant i16 2000
@i32 = internal constant i32 300000
@i64 = internal constant i64 4000000000
@i128 = internal constant i128 50000000000000000000
@i81 = internal constant i8 10
@i162 = internal global i16 0
@i323 = internal global i32 0
@i644 = internal global i64 0
@i1285 = internal global i128 0
@u8 = internal constant i8 10
@u16 = internal constant i16 2000
@u32 = internal constant i32 300000
@u64 = internal constant i64 4000000000
@u128 = internal constant i128 50000000000000000000
@u86 = internal constant i8 10
@u167 = internal global i16 0
@u328 = internal global i32 0
@u649 = internal global i64 0
@u12810 = internal global i128 0
@i = internal constant i128 0
@i11 = internal global i128 0
@i12 = internal global i128 0
@i13 = internal global i128 0
@i14 = internal global i128 0
@i15 = internal global i128 0
@i17 = internal global i128 0
@i18 = internal global i128 0
@i19 = internal global i128 0
@i20 = internal global i128 0
@i21 = internal global i128 0

define internal void @integer..init() {
  %1 = load i8* @i81
  %2 = sext i8 %1 to i16
  store i16 %2, i16* @i162
  %3 = load i16* @i162
  %4 = sext i16 %3 to i32
  store i32 %4, i32* @i323
  %5 = load i32* @i323
  %6 = sext i32 %5 to i64
  store i64 %6, i64* @i644
  %7 = load i64* @i644
  %8 = sext i64 %7 to i128
  store i128 %8, i128* @i1285
  %9 = load i8* @u86
  %10 = zext i8 %9 to i16
  store i16 %10, i16* @u167
  %11 = load i16* @u167
  %12 = zext i16 %11 to i32
  store i32 %12, i32* @u328
  %13 = load i32* @u328
  %14 = zext i32 %13 to i64
  store i64 %14, i64* @u649
  %15 = load i64* @u649
  %16 = zext i64 %15 to i128
  store i128 %16, i128* @u12810
  %17 = load i8* @i81
  %18 = sext i8 %17 to i128
  store i128 %18, i128* @i11
  %19 = load i16* @i162
  %20 = sext i16 %19 to i128
  store i128 %20, i128* @i12
  %21 = load i32* @i323
  %22 = sext i32 %21 to i128
  store i128 %22, i128* @i13
  %23 = load i64* @i644
  %24 = sext i64 %23 to i128
  store i128 %24, i128* @i14
  %25 = load i128* @i1285
  store i128 %25, i128* @i15
  %26 = load i8* @u86
  %27 = zext i8 %26 to i128
  store i128 %27, i128* @i17
  %28 = load i16* @u167
  %29 = zext i16 %28 to i128
  store i128 %29, i128* @i18
  %30 = load i32* @u328
  %31 = zext i32 %30 to i128
  store i128 %31, i128* @i19
  %32 = load i64* @u649
  %33 = zext i64 %32 to i128
  store i128 %33, i128* @i20
  %34 = load i128* @u12810
  store i128 %34, i128* @i21
  ret void
}

define i32 @main(i32, i8**, i8**) {
  call void @integer..init()
  ret i32 0
}
